plugin/src/main/java/org/autorefactor/refactoring/rules/WakeLockRefactoring.java

```diff
@@ -0,0 +1,208 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program under LICENSE-GNUGPL.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution under LICENSE-ECLIPSE, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.autorefactor.refactoring.rules;
+
+import static org.autorefactor.refactoring.ASTHelper.DO_NOT_VISIT_SUBTREE;
+import static org.autorefactor.refactoring.ASTHelper.VISIT_SUBTREE;
+import static org.autorefactor.refactoring.ASTHelper.isMethod;
+
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.Refactorings;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ASTVisitor;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.IfStatement;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.Modifier;
+import org.eclipse.jdt.core.dom.Modifier.ModifierKeyword;
+import org.eclipse.jdt.core.dom.NormalAnnotation;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.SuperMethodInvocation;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+
+/** See {@link #getDescription()} method. */
+public class WakeLockRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        // TODO Auto-generated method stub
+        return "Failing to release a wakelock properly can keep the Android device in a high "
+            + "power mode, which reduces battery life. There are several causes of this, such "
+            + "as releasing the wake lock in onDestroy() instead of in onPause(), failing to "
+            + "call release() in all possible code paths after an acquire(), and so on.";
+    }
+
+    @Override
+    public String getName() {
+        return "WakeLockRefactoring";
+    }
+
+    @Override
+    public boolean visit(MethodInvocation node) {
+        if (isMethod(node, "android.os.PowerManager.WakeLock", "release")) {
+            // check whether it is being called in onDestroy
+            final Refactorings r = this.ctx.getRefactorings();
+            final ASTBuilder b = this.ctx.getASTBuilder();
+            MethodDeclaration enclosingMethod;
+            enclosingMethod = (MethodDeclaration) ASTNodes.getParent(node, ASTNode.METHOD_DECLARATION);
+            if (isMethod(enclosingMethod.resolveBinding(), "android.app.Activity", "onDestroy")) {
```

Comment:
Actually I have just merged this change (along with another change useful to you).
Please rebase.